packages/instructions/src/__typetests__/instruction-typetest.ts

```diff
@@ -44,3 +48,20 @@ import {
         IInstructionWithAccounts<readonly (IAccountMeta | IAccountLookupMeta)[]> &
         IInstructionWithData<Uint8Array>;
 }
+
+// narrowing to known address
+{
+    const instruction = {} as unknown as IInstruction;
+    const myAddress = '1111' as Address<'1111'>;
+    type MyAddress = typeof myAddress;
```

Comment:
I think this is meant to be `type MyAddress = '1111'` otherwise you end up with `IInstruction<MyAddress>` producing `Address<Address<'1111'>>`

Reply:
This was actually caused by a typo when I tweaked the function

With that fixed both work correctly:
```ts
instruction satisfies IInstruction<Address<'1111>>;
instruction satisfies IInstruction<'1111'>;
```